ironfish/src/rpc/routes/wallet/addKnownTransactions.ts

```diff
@@ -0,0 +1,249 @@
+
+    if (!startHeader || !(await context.chain.isHeadChain(startHeader))) {
+      throw new RpcValidationError('Start block is not on the head chain.')
+    }
+
+    if (!endHeader || !(await context.chain.isHeadChain(endHeader))) {
+      throw new RpcValidationError('End block is not on the head chain.')
+    }
+
+    if (startHeader.sequence > endHeader.sequence) {
+      throw new RpcValidationError('End block must be greater than or equal to start block.')
+    }
+
+    // Validate account state
+    const account = getAccount(context.wallet, request.data.account)
+
+    if (account.scanningEnabled) {
+      throw new RpcResponseError(
+        `Cannot add known transactions while account syncing is enabled. Try calling wallet/stopScanning first.`,
+        RPC_ERROR_CODES.ERROR,
+        409,
+      )
+    }
+
+    // Validate account head is compatible with start and end blocks
+    let accountHead = await account.getHead()
+
+    if (accountHead !== null) {
+      const accountHeader = await context.chain.getHeader(accountHead.hash)
+      if (!accountHeader) {
+        throw new Error(`accountHead ${accountHead.hash.toString('hex')} not found in chain`)
+      }
+
+      const fork = await context.chain.findFork(startHeader, accountHeader)
+
+      // If fork is startHeader:
+      //  - This means startHeader is linearly older than accountHead.
+      //  - We will rewind accountHead to the block before startHeader.
+      //  - You could also ignore all blocks before and including accountHead. (Note that this also
+      //    applies if startHeader == accountHead). We choose not to because it keeps the code simpler
+      //    for now.
+      // If fork is accountHead or neither:
+      //  - This means accountHead is earlier than startHeader.
+      //  - we want to abort if there's a gap between accountHead and startHeader, since there may be
+      //    transactions in the gap relevant to the account. (alternatively, we could consider scanning
+      //    all blocks in the gap, as in a typical wallet scanß)
+      //  - If fork is startHeader.previousBlockHash, the fork directly connects to startHeader, so
+      //    rewind accountHead to startHeader.previousBlockHash. Note that we must rewind, we cannot
+      //    ignore blocks in this case because the accountHead is on a different chain.
+      //  - Otherwise there's a gap between accountHead and startHeader, so abort
+      if (!fork.equals(startHeader) && !fork.hash.equals(startHeader.previousBlockHash)) {
+        const nextHash = (await context.chain.getNextHash(fork.hash)) ?? fork.hash
+        throw new RpcValidationError(`Start must be ${nextHash?.toString('hex')} or earlier.`)
+      }
+
+      while (accountHead !== null && !accountHead.hash.equals(startHeader.previousBlockHash)) {
+        const header: BlockHeader | null = await context.chain.getHeader(accountHead.hash)
```

Comment:
Is this a typescript bug? I cannot figure out why TS interprets this type as `any` without this type annotation.